### ISSUE TITLE
Add ReduceMean

### ISSUE THREAD

**Issue by cesarsouza** [migueldeicaza/TensorFlowSharp#118]
It is a simple operation, but one that is missing nonetheless. The implementation should be just like:

```csharp
/// <summary>
/// Computes the mean of elements across dimensions of a tensor.
/// </summary>
/// <returns>The reduced tensor.</returns>
/// <param name="input">The tensor to reduce. Should have numeric type.</param>
/// <param name="axis">The dimensions to reduce. If not se (the default), reduces all dimensions.</param>
/// <param name="keep_dims">If set to <c>true</c> retains reduced dimensions with length 1.</param>
/// <param name="operName">A name for the operation, optional.</param>
/// <remarks>
/// <para>
///   Reduces input_tensor along the dimensions given in axis.
/// Unless keep_dims is true, the rank of the tensor is reduced by 1 for each
/// entry in axis. If keep_dims is true, the reduced dimensions
/// are retained with length 1.</para>
/// 
/// <para>
/// If axis has no entries, all dimensions are reduced, and a
/// tensor with a single element is returned.</para>
/// </remarks>
public TFOutput ReduceMean (TFOutput input, TFOutput? axis = null, bool? keep_dims = false, string operName = null)
{
	return Mean (input, this.ReduceDims (input, axis), keep_dims, operName);
}
```

I will be adding a PR for this implementation with proper unit tests in a minute.

Regards,
Cesar

**Comment by cesarsouza** [comment]
Oh, actually, I might need some of the methods that I had introduced in #117. I will send the new PR once that PR is integrated (or rework it in case it is rejected).

**Comment by cesarsouza** [comment]
Merged in #126
